torch/lib/c10d/reducer_cuda.cpp

```diff
@@ -0,0 +1,91 @@
+#include <c10d/reducer.hpp>
+
+#ifdef USE_CUDA
+
+#include <c10/core/DeviceGuard.h>
+#include <ATen/cuda/CUDAEvent.h>
+
+namespace c10d {
+namespace {
+
+const int kMilliSecondToNanosSecond = 1000000;
+
+class CudaTimer : public Timer {
+ private:
+  c10::Device device;
+
+  at::cuda::CUDAEvent forward_start = at::cuda::CUDAEvent(cudaEventDefault);
+  at::cuda::CUDAEvent backward_compute_start =
+      at::cuda::CUDAEvent(cudaEventDefault);
+  at::cuda::CUDAEvent backward_compute_end =
+      at::cuda::CUDAEvent(cudaEventDefault);
+  at::cuda::CUDAEvent backward_comm_start =
+      at::cuda::CUDAEvent(cudaEventDefault);
+  at::cuda::CUDAEvent backward_comm_end =
+      at::cuda::CUDAEvent(cudaEventDefault);
+
+  at::cuda::CUDAEvent& getEvent(Event event) {
+    switch (event) {
+      case Event::kForwardStart:
+        return forward_start;
+      case Event::kBackwardComputeStart:
+        return backward_compute_start;
+      case Event::kBackwardComputeEnd:
+        return backward_compute_end;
+      case Event::kBackwardCommStart:
+        return backward_comm_start;
+      case Event::kBackwardCommEnd:
+        return backward_comm_end;
+      default:
+        TORCH_INTERNAL_ASSERT(false);
+    }
+    // In case the compiler doesn't get it.
+    return forward_start;
```

Comment:
could you get a warning/lint on unreachable code instead?

Reply:
Dunno, it doesn't seem we're getting any such lints from clang-tidy though? But let me remove this if you prefer